### ISSUE TITLE
Sam Package Requires S3 Bucket For PackageType: Image

### ISSUE THREAD

**Issue by metaskills** [aws/aws-sam-cli#2450]
I can confirm that the `--s3-bucket` and `--s3-prefix` have no use during `sam package` when using the `PackageType: Image` process with the `--image-repository` option. However when I remove the S3 options, the command won't run. Can we remove that if indeed that is an oversight?

**Comment by sriram-mv** [comment]
I'm unable to reproduce this. I just deployed a pure image stack with just the following command after running sam build for the sam init template for python3.8 for Images.

Template after running sam build

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Description: 'python3.8

  Sample SAM Template for python-image-app

  '
Globals:
  Function:
    Timeout: 3
Resources:
  HelloWorldFunction:
    Type: AWS::Serverless::Function
    Properties:
      PackageType: Image
      Events:
        HelloWorld:
          Type: Api
          Properties:
            Path: /hello
            Method: get
      ImageUri: helloworldfunction:python3.8-v1
    Metadata:
      Dockerfile: Dockerfile
      DockerContext: ./hello_world
      DockerTag: python3.8-v1
Outputs:
  HelloWorldApi:
    Description: API Gateway endpoint URL for Prod stage for Hello World function
    Value:
      Fn::Sub: https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/hello/
  HelloWorldFunction:
    Description: Hello World Lambda Function ARN
    Value:
      Fn::GetAtt:
      - HelloWorldFunction
      - Arn
  HelloWorldFunctionIamRole:
    Description: Implicit IAM Role created for Hello World function
    Value:
      Fn::GetAtt:
      - HelloWorldFunctionRole
      - Arn
```


```
sam package --image-repository <uri>
```

```
The push refers to repository [<>.dkr.ecr.us-east-1.amazonaws.com/<>]
e5eaa995d73e: Layer already exists
65241a26bb87: Layer already exists
c25cc523d48d: Layer already exists
b16d3076cef9: Pushing [==========================>                        ]  104.1MB/199.7MB
d6fa53d6caa6: Layer already exists
d6ad3c04b778: Layer already exists
3ae9cc8c3585: Layer already exists
af6d16f2417e: Layer already exists
helloworldfunction-f8d81feadd10-python3.8-v1: digest: sha256:e34e3fd0048a3e8ba4d131d40889fb429d054f47d25a1e0c4eacfa834420e45c size: 1999

AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Description: 'python3.8

  Sample SAM Template for python-image-app

  '
Globals:
  Function:
    Timeout: 3
Resources:
  HelloWorldFunction:
    Type: AWS::Serverless::Function
    Properties:
      PackageType: Image
      Events:
        HelloWorld:
          Type: Api
          Properties:
            Path: /hello
            Method: get
      ImageUri:<>.dkr.ecr.us-east-1.amazonaws.com/sam-cli:helloworldfunction-f8d81feadd10-python3.8-v1
    Metadata:
      Dockerfile: Dockerfile
      DockerContext: ./hello_world
      DockerTag: python3.8-v1
Outputs:
  HelloWorldApi:
    Description: API Gateway endpoint URL for Prod stage for Hello World function
    Value:
      Fn::Sub: https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/hello/
  HelloWorldFunction:
    Description: Hello World Lambda Function ARN
    Value:
      Fn::GetAtt:
      - HelloWorldFunction
      - Arn
  HelloWorldFunctionIamRole:
    Description: Implicit IAM Role created for Hello World function
    Value:
      Fn::GetAtt:
      - HelloWorldFunctionRole
      - Arn
```

**Comment by prenx4x** [comment]
I tried to repro this, but found no issues without s3-bucket, s3-prefix or resolve-s3. Can you provide more details/steps to reproduce this?

**Comment by metaskills** [comment]
Yup, still seeing it. My `sam package` command is this.

```
sam package \
  --region "$AWS_DEFAULT_REGION" \
  --template-file ./.aws-sam/build/template.yaml \
  --output-template-file ./.aws-sam/build/packaged.yaml \
  --image-repository "$IMAGE_REPOSITORY"
```

And the error is this.

```
== SAM package...
Usage: sam package [OPTIONS]
Try 'sam package --help' for help.

Error: Missing option '--s3-bucket'
```

I'm using this cookiecutter. https://github.com/customink/lamby-cookiecutter/tree/master/%7B%7Bcookiecutter.project_name%7D%7D

**Comment by metaskills** [comment]
I'm not sure why this is, but I found the difference and what causes this to happen. You can too if you change your template to include a API Gateway Resource like this.

```yaml
MyHttpApi:
  Type: AWS::Serverless::HttpApi
  Properties:
    StageName: staging
```

And have your Lambda event use this.

```yaml
Events:
  MyHttpApiProxy:
    Type: HttpApi
    Properties:
      ApiId: !Ref MyrHttpApi
```

This will result in the S3 Bucket error.

**Comment by bjfletcher** [comment]
It's a bug in the `aws-sam-cli` code.

I've found that if `template.yaml` contains just `AWS::Serverless::Function` resources with `PackageType: Image` then it doesn't complain asking for `s3-bucket`.

If however `template.yaml` contains non-function resources (e.g. `AWS::Events::Rule`) without `PackageType: Image` then it complains asking for `s3-bucket`. When I add `PackageType: Image` to these non-function resources, then it doesn't complain asking for `s3-bucket`. (But `sam deploy` will then point out that these non-function resources shouldn't have `PackageType: Image`.)

In short, `aws-sam-cli` code needs to be fixed to check `PackageType` **only** in the `AWS::Serverless::Function` resources.

**Comment by metaskills** [comment]
Interesting, and thank you! In my case I have this additional resource.

```yaml
  RailsHttpApi:
    Type: AWS::Serverless::HttpApi
    Properties:
      StageName: !Ref RailsEnv
```

Which does not need an S3 bucket. So agreed on your assessment.

**Comment by halilodun** [comment]
hello
i need to talk someone

**Comment by andreswebs** [comment]
I confirm the behavior described by @bjfletcher

**Comment by aahung** [comment]
released in 1.22.0, let us know if the problem still exists

**Comment by sarathyplkr5** [comment]
Hi,
Im getting an error when using sam package.
Error: Cannot use both --resolve-s3 and --s3-bucket parameters. Please use only one.

I am trying a basic "hello world" application.

**Comment by aahung** [comment]
@sarathyplkr5 can you open an issue with more detail? This issue has been closed for a while.

**Comment by RLain** [comment]
Old thread but wanted to share insights in case it helps others (Also doing a basic hello-world from Stephane's DVA-02 course).

**Location**: Using CloudShell.

The `sam package` command did `not` work:
$ sam package --template-file template.yaml --output-template-file packaged.yaml --s3-bucket name-of-bucket

```bash
[cloudshell-user@ip-10-6-91-153 sam-sar-repo]$ sam package --template-file template.yaml --output-template-file packaged.yaml --s3-bucket becca-sar-demo-2023
Error: Cannot use both --resolve-s3 and --s3-bucket parameters. Please use only one.
```

```bash
The `aws cloudformation package` command `does` work:
[cloudshell-user@ip-10-6-91-153 sam-sar-repo]$ aws cloudformation package \
> --template-file template.yaml \
> --output-template-file packaged.yaml \
> --s3-bucket becca-sar-demo-2023 \
> 

Successfully packaged artifacts and wrote output template to file packaged.yaml.
Execute the following command to deploy the packaged template
aws cloudformation deploy --template-file /home/cloudshell-user/sam-sar-repo/packaged.yaml --stack-name <YOUR STACK NAME>
```

Feels odd. Wanted to share in case I am missing anything obvious/this may help others.
